octodns/provider/digitalocean.py

```diff
@@ -0,0 +1,333 @@
+#
+#
+#
+
+from __future__ import absolute_import, division, print_function, \
+    unicode_literals
+
+from collections import defaultdict
+from requests import Session
+import logging
+
+from ..record import Record
+from .base import BaseProvider
+
+
+class DigitalOceanClientException(Exception):
+    pass
+
+
+class DigitalOceanClientNotFound(DigitalOceanClientException):
+
+    def __init__(self):
+        super(DigitalOceanClientNotFound, self).__init__('Not Found')
+
+
+class DigitalOceanClientUnauthorized(DigitalOceanClientException):
+
+    def __init__(self):
+        super(DigitalOceanClientUnauthorized, self).__init__('Unauthorized')
+
+
+class DigitalOceanClient(object):
+    BASE = 'https://api.digitalocean.com/v2'
+
+    def __init__(self, token):
+        sess = Session()
+        sess.headers.update({'Authorization': 'Bearer {}'.format(token)})
+        self._sess = sess
+
+    def _request(self, method, path, params=None, data=None):
+        url = '{}{}'.format(self.BASE, path)
+        resp = self._sess.request(method, url, params=params, json=data)
+        if resp.status_code == 401:
+            raise DigitalOceanClientUnauthorized()
+        if resp.status_code == 404:
+            raise DigitalOceanClientNotFound()
+        resp.raise_for_status()
+        return resp
+
+    def domain(self, name):
+        path = '/domains/{}'.format(name)
+        return self._request('GET', path).json()
+
+    def domain_create(self, name):
+        path = '/domains'
+
+        # Digitalocean requires an IP on zone creation
+        resp = self._request('POST', path, data={'name': name,
+                                                 'ip_address': '127.0.0.1'})
+
+        # After the zone is created, immeadiately delete the record
+        records = self.records(name)
```

Comment:
Weird, is that what the IP address above is used for? It just stick an A-record on the top? Weird.

If we're sure there's only going to be a single record here it could probably short-cut things a bit and do something like:

```python
self.record_delete(name, records[0]['id'])
```

What's there is a little safer in the case that there's more than one so it's not a problem.

Reply:
DigitalOcean automatically put's in their NS records, so most likely the A record will be `records[3]['id']`, but you can't guarantee what order the apex A record will be at. Only sure way is to search for them and delete them when found.

It is a very weird thing to require, especially when it's not required in the UI.